src/main/scala/firrtl/passes/MemUtils.scala

```diff
@@ -64,71 +63,67 @@ object toBits {
-  def widthOf(dt: Type): BigInt = dt match {
+  private def widthOf(dt: Type): BigInt = dt match {
     case t: VectorType => t.size * bitWidth(t.tpe)
     case t: BundleType => t.fields.map(f => bitWidth(f.tpe)).foldLeft(BigInt(0))(_+_)
     case UIntType(IntWidth(width)) => width
     case SIntType(IntWidth(width)) => width
-    case t => error("Unknown type encountered in bitWidth!")
+    case _ => error("Unknown type encountered in bitWidth!")
   }
 }
 
 object fromBits {
   def apply(lhs: Expression, rhs: Expression): Statement = {
     val fbits = lhs match {
-      case ex: WRef => getPart(ex, ex.tpe, rhs, 0)
-      case ex: WSubField => getPart(ex, ex.tpe, rhs, 0)
-      case ex: WSubIndex => getPart(ex, ex.tpe, rhs, 0)
-      case t => error("Invalid LHS expression for fromBits!")
+      case ex @ (_: WRef | _: WSubField | _: WSubIndex) => getPart(ex, ex.tpe, rhs, 0)
+      case _ => error("Invalid LHS expression for fromBits!")
     }
     Block(fbits._2)
   }
-  def getPartGround(lhs: Expression, lhst: Type, rhs: Expression, offset: BigInt): (BigInt, Seq[Statement]) = {
+  private def getPartGround(lhs: Expression, lhst: Type, rhs: Expression,
+      offset: BigInt): (BigInt, Seq[Statement]) = {
     val intWidth = bitWidth(lhst)
-    val sel = DoPrim(PrimOps.Bits, Seq(rhs), Seq(offset+intWidth-1, offset), UnknownType)
+    val sel = DoPrim(PrimOps.Bits, Seq(rhs), Seq(offset + intWidth - 1, offset), UnknownType)
     (offset + intWidth, Seq(Connect(NoInfo, lhs, sel)))
   }
-  def getPart(lhs: Expression, lhst: Type, rhs: Expression, offset: BigInt): (BigInt, Seq[Statement]) = {
-    lhst match {
-      case t: VectorType => {
-        var currentOffset = offset
-        var stmts = Seq.empty[Statement]
-        for (i <- (0 until t.size)) {
-          val (tmpOffset, substmts) = getPart(WSubIndex(lhs, i, t.tpe, UNKNOWNGENDER), t.tpe, rhs, currentOffset)
-          stmts = stmts ++ substmts
-          currentOffset = tmpOffset
-        }
-        (currentOffset, stmts)
-      }
-      case t: BundleType => {
-        var currentOffset = offset
-        var stmts = Seq.empty[Statement]
-        for (f <- t.fields.reverse) {
-          val (tmpOffset, substmts) = getPart(WSubField(lhs, f.name, f.tpe, UNKNOWNGENDER), f.tpe, rhs, currentOffset)
-          stmts = stmts ++ substmts
-          currentOffset = tmpOffset
-        }
-        (currentOffset, stmts)
-      }
-      case t: GroundType => getPartGround(lhs, t, rhs, offset)
-      case t => error("Unknown type encountered in fromBits!")
+  private def getPart(lhs: Expression, lhst: Type, rhs: Expression,
+      offset: BigInt): (BigInt, Seq[Statement]) = lhst match {
+    case t: VectorType => (0 until t.size foldRight (offset, Seq[Statement]())){ case (i, (curOffset, stmts)) =>
+      val subidx = WSubIndex(lhs, i, t.tpe, UNKNOWNGENDER)
+      val (tmpOffset, substmts) = getPart(subidx, t.tpe, rhs, curOffset)
+      (tmpOffset, stmts ++ substmts)
+    }
```

Comment:
There is way too much going on the line of the case, would prefer something like:
```scala
    case t: VectorType =>
      (0 until t.size).foldRight(offset, Seq[Statement]()) { 
        case (i, (curOffset, stmts)) =>
          val subidx = WSubIndex(lhs, i, t.tpe, UNKNOWNGENDER)
          val (tmpOffset, substmts) = getPart(subidx, t.tpe, rhs, curOffset)
          (tmpOffset, stmts ++ substmts)
      }
```
I also actually prefer calling folds with `.` anyway because it helps keep the number of parentheses down, that's not a huge deal but I think splitting up the lines a bit is